game-core/src/main/java/games/strategy/engine/data/Unit.java

```diff
@@ -81,10 +81,8 @@ public boolean equals(final Object o) {
   }
 
   public boolean isEquivalent(final Unit unit) {
-    if (m_type == null || m_owner == null) {
-      return false;
-    }
-    return m_type.equals(unit.getType()) && m_owner.equals(unit.getOwner()) && m_hits == unit.getHits();
+    return m_type != null && m_owner != null && m_type.equals(unit.getType()) && m_owner.equals(unit.getOwner())
```

Comment:
could you group this up so `m_owner` NPE check is close to its usage. 

Any idea why `unit` is not checked for null?

Reply:
No Idea